cmd/queue/main.go

```diff
@@ -277,7 +306,13 @@ func probeQueueHealthPath(port int, timeoutSeconds int) error {
 			return false, nil
 		}
 		defer res.Body.Close()
-		return health.IsHTTPProbeReady(res), nil
+		success := health.IsHTTPProbeReady(res)
+		// The check for preferForScaledown() fails readiness faster
+		// in the presence of the label
+		if !success && preferPodForScaledown(logger, env.DownwardAPILabelsPath) {
```

Comment:
Do we even need the entire check in the presence of `preferPodForScaledown`? It feels like we can just put an `if preferPodForScaledown` at the top of the function and return immediately if `true`? We don't need to bother checking the container in that case I think?

Reply:
i am not super keen on the idea of failing the binary that does the probing to indicate that the container is unhealthy. It kind breaks the definition of the health endpoint in my mind. Particularly if at some point we choose to move away from using the custom binary we have for probing.

Reply:
But aren't we failing it with this check anyway? We'd just fail it early because we already know that the label is set before even making an HTTP call?

Reply:
it still relies on `success == false` tho. ideally we didn't have to do the check for `preferPodForScaledown` in the binary if we didnt have `PollImmediateUntil()`. `PollImmediateUntil` is a very aggressive way of ensuring readiness, but I guess that had to be done for some resiliency. In an ideal world it would have only been `success == false` and done.

Reply:
I fail to see how this works. Since if `sucsess==true`, which _will most likely be the case_, since we're operating normally so far, we'll never even check the file?

Reply:
checking the file happens in `knativeProbeHandler` and success is determined based on the status code returned by the server

Reply:
OK, reading more; it seems that this code is executed only in the `standalone probe` to check QP health. Do we even care about this path?

Reply:
we shouldn't, except we do. For [this reason](https://github.com/knative/serving/pull/6552#discussion_r373151803), i.e the one before last comment of mine.